flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/ParquetTableSource.java

```diff
@@ -200,6 +204,8 @@ private ParquetTableSource(String path, MessageType parquetSchema, Configuration
 			for (FilterPredicate converted : convertedPredicates.subList(1, convertedPredicates.size())) {
 				parquetPredicate = FilterApi.and(parquetPredicate, converted);
 			}
+			parquetPredicate = LogicalInverseRewriter.rewrite(parquetPredicate);
+			SchemaCompatibilityValidator.validate(parquetPredicate, parquetSchema);
```

Comment:
Why we need rewrite and validate?

Reply:
It was added when i do local test. it is not needed. Reverted back.